src/cli/NativeMsgIface.cpp

```diff
@@ -0,0 +1,430 @@
+
+// Explicit deduction guide (not needed as of C++20)
+template<class... Ts> overloaded(Ts...) -> overloaded<Ts...>;
+
+std::shared_ptr<marian::Options> makeOptions(const std::string &path_to_model_dir, const translateLocally::marianSettings &settings) {
+    std::shared_ptr<marian::Options> options(marian::bergamot::parseOptionsFromFilePath(path_to_model_dir + "/config.intgemm8bitalpha.yml"));
+    options->set("cpu-threads", settings.cpu_threads,
+                 "workspace", settings.workspace,
+                 "mini-batch-words", 1000,
+                 "alignment", "soft",
+                 "quiet", true);
+    return options;
+}
+
+// Little helper function that sets up a SingleShot connection in both Qt 5 and 6
+template <typename Sender, typename Emitter, typename Slot, typename... Args>
+QMetaObject::Connection connectSingleShot(Sender *sender, void (Emitter::*signal)(Args ...args), const QObject *context, Slot slot) {
+#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
+    std::shared_ptr<QMetaObject::Connection> connection = std::make_shared<QMetaObject::Connection>();
+    return *connection = QObject::connect(sender, signal, context, [=](Args ...args) {
+        QObject::disconnect(*connection);
+        slot(std::forward<Args>(args)...);
+    });
+#else
+    return QObject::connect(sender, signal, context, slot, Qt::SingleShotConnection);
+#endif
+}
+
+// Little helper to print QSet<QString> and QList<QString> without the need to
+// convert them into a QStringList.
+template <typename T>
+QString join(QString glue, T const &list) {
+    QString out;
+    for (auto &&item : list) {
+        if (!out.isEmpty())
+            out += glue;
+        out += item;
+    }
+    return out;
+};
+
+}
+
+NativeMsgIface::NativeMsgIface(QObject * parent) :
+      QObject(parent)
+      , network_(this)
+      , settings_(this)
+      , models_(this, &settings_)
+      , operations_(0)
+      , die_(false)
+    {
+    qRegisterMetaType<std::shared_ptr<std::vector<char>>>("std::shared_ptr<std::vector<char>>");
+    
+    // Disable synchronisation with C style streams. That should make IO faster
+    std::ios_base::sync_with_stdio(false);
+
+    // Init the marian translation service:
+    marian::bergamot::AsyncService::Config serviceConfig;
+    serviceConfig.numWorkers = settings_.marianSettings().cpu_threads;
+    serviceConfig.cacheSize = settings_.marianSettings().translation_cache ? kTranslationCacheSize : 0;
+    service_ = std::make_shared<marian::bergamot::AsyncService>(serviceConfig);
+
+    // Pick up on network errors
+    // TODO: not sure if every request always returns on 1 error. fetchRemoteModels for example
+    // will cut short if a request is already in progress. If that one fails, the second request
+    // that got cut short will have no response, and stay pending.
+    connect(&network_, &Network::error, this, [&](QString err, QVariant data) {
+        if (data.canConvert<Request>())
```

Comment:
I think we should have an else branch that writes to stderr. The else branch is a programmer error in this case.